# HG changeset patch
# Parent  b711c3a217123f374b86a328d7e297290a726aab
Rename PULSE to SQUARE and document set_waveform's skew_ratio argument

I think SQUARE is the convention in this context.

diff --git a/core/proto.c b/core/proto.c
--- a/core/proto.c
+++ b/core/proto.c
@@ -156,7 +156,7 @@
     assert(saturation >= 0 && saturation <= UINT16_MAX);
     assert(brightness >= 0 && brightness <= UINT16_MAX);
     assert(kelvin >= 2500 && kelvin <= 9000);
-    assert(waveform <= LGTD_LIFX_WAVEFORM_PULSE);
+    assert(waveform <= LGTD_LIFX_WAVEFORM_SQUARE);
     assert(skew_ratio >= -32767 && skew_ratio <= 32768);
     assert(period >= 0);
     assert(cycles >= 0);
diff --git a/docs/protocol.rst b/docs/protocol.rst
--- a/docs/protocol.rst
+++ b/docs/protocol.rst
@@ -52,7 +52,7 @@
 .. function:: set_waveform(target, waveform, h, s, b, k, period, cycles, skew_ratio, transient)
 
    :param string waveform: One of ``SAW``, ``SINE``, ``HALF_SINE``,
-                           ``TRIANGLE``, ``PULSE``.
+                           ``TRIANGLE``, ``SQUARE``.
    :param float h: Hue from 0 to 360.
    :param float s: Saturation from 0 to 1.
    :param float b: Brightness from 0 to 1.
@@ -64,6 +64,23 @@
                           end of the waveform, otherwise it will revert back to
                           its original state.
 
+   The meaning of the ``skew_ratio`` argument depends on the type of waveform:
+
+   +---------------+-----------------------------------------------------------+
+   | ``SAW``       | Should be 0.5.                                            |
+   +---------------+-----------------------------------------------------------+
+   | ``SINE``      | Defines the peak point of the function, 0.5 gives you a   |
+   |               | sine and 1 or 0 will give you cosine. Ignored by firmware |
+   |               | 1.1.                                                      |
+   +---------------+-----------------------------------------------------------+
+   | ``HALF_SINE`` | Should be 0.5.                                            |
+   +---------------+-----------------------------------------------------------+
+   | ``TRIANGLE``  | Defines the peak point of the function like ``SINE``.     |
+   |               | Ignored by firmware 1.1.                                  |
+   +---------------+-----------------------------------------------------------+
+   | ``SQUARE``    | Ratio of a cycle the targets are set to the given color.  |
+   +---------------+-----------------------------------------------------------+
+
 .. function:: get_light_state(target)
 
     Return a list of dictionnaries, each dict representing the state of one
diff --git a/examples/lightsc.py b/examples/lightsc.py
--- a/examples/lightsc.py
+++ b/examples/lightsc.py
@@ -81,10 +81,10 @@
     )
 
 
-def pulse(socket, target, h, s, b, k, period, cycles,
-          duty_cycle=0.5, transient=True):
+def square(socket, target, h, s, b, k, period, cycles,
+           duty_cycle=0.5, transient=True):
     return set_waveform(
-        socket, target, "PULSE", h, s, b, k,
+        socket, target, "SQUARE", h, s, b, k,
         cycles=cycles,
         period=period,
         skew_ratio=duty_cycle,
diff --git a/lifx/wire_proto.c b/lifx/wire_proto.c
--- a/lifx/wire_proto.c
+++ b/lifx/wire_proto.c
@@ -444,7 +444,7 @@
     WAVEFORM_ENTRY("SINE"),
     WAVEFORM_ENTRY("HALF_SINE"),
     WAVEFORM_ENTRY("TRIANGLE"),
-    WAVEFORM_ENTRY("PULSE"),
+    WAVEFORM_ENTRY("SQUARE"),
     WAVEFORM_ENTRY("INVALID")
 };
 
diff --git a/lifx/wire_proto.h b/lifx/wire_proto.h
--- a/lifx/wire_proto.h
+++ b/lifx/wire_proto.h
@@ -223,7 +223,7 @@
     LGTD_LIFX_WAVEFORM_SINE = 1,
     LGTD_LIFX_WAVEFORM_HALF_SINE = 2,
     LGTD_LIFX_WAVEFORM_TRIANGLE = 3,
-    LGTD_LIFX_WAVEFORM_PULSE = 4,
+    LGTD_LIFX_WAVEFORM_SQUARE = 4,
     LGTD_LIFX_WAVEFORM_INVALID = 5,
 };
 
diff --git a/tests/lifx/wire_proto/test_wire_proto_waveform_table.c b/tests/lifx/wire_proto/test_wire_proto_waveform_table.c
--- a/tests/lifx/wire_proto/test_wire_proto_waveform_table.c
+++ b/tests/lifx/wire_proto/test_wire_proto_waveform_table.c
@@ -27,9 +27,9 @@
         errx(1, "Expected WAVEFORM_TRIANGLE");
     }
 
-    rv = lgtd_lifx_wire_waveform_string_id_to_type("PULSE", 5);
-    if (rv != LGTD_LIFX_WAVEFORM_PULSE) {
-        errx(1, "Expected WAVEFORM_PULSE");
+    rv = lgtd_lifx_wire_waveform_string_id_to_type("SQUARE", 6);
+    if (rv != LGTD_LIFX_WAVEFORM_SQUARE) {
+        errx(1, "Expected WAVEFORM_SQUARE");
     }
 
     rv = lgtd_lifx_wire_waveform_string_id_to_type("TEST", 4);
